jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py

```diff
@@ -258,8 +294,12 @@ def collapse_to_spectrum(self, add_data=True, **kwargs):
         if self.conflicting_aperture_and_function:
             raise ValueError(self.conflicting_aperture_error_message)
 
-        spectral_cube = self._app._jdaviz_helper._loaded_flux_cube
-        uncert_cube = self._app._jdaviz_helper._loaded_uncert_cube
+        spectral_cube = self.dataset.selected_dc_item
+        if self.dataset.selected == self._app._jdaviz_helper._loaded_flux_cube.label:
+            uncert_cube = self._app._jdaviz_helper._loaded_uncert_cube
+        else:
+            # TODO: allow selecting or associating an uncertainty cube?
```

Comment:
Is this TODO a different ticket somewhere?

Reply:
not yet, but agreed we should create it/them.  Let's decide on what behavior we want now and for the future and then I can create any necessary tickets.  

I had to generalize spectral extraction to technically allow any input flux cube, rather than just the one loaded into the app, although this isn't yet exposed to the user in the UI.  Do we want that flexibility either now or in the future (so the user could extract from a smoothed cube or a model-fit cube, etc)?

And if so, then should we always assume the original input uncertainty cube or have a separate selection for that?

Reply:
Is it possible to reuse the "association" logic from Brett's DQ work?

Reply:
hmmm, maybe we can use something like that.  That currently is designed so that the "child" requires the "parent" to be visible first, which isn't what we want for the uncertainty cube.  But somehow linking them internally would definitely be useful.  But is it too restrictive?  Would you ever want to be able to use the original uncertainty cube on a model fit or something?

Reply:
> Would you ever want to be able to use the original uncertainty cube on a model fit or something?

If you mean associating the original uncertainty with the _result_ from model fit, maybe but since we have full control of the model fitting plugin, surely we can make it happen behind the scene with Brett's thingy?

ref: https://docs.astropy.org/en/stable/modeling/example-fitting-line.html#fit-using-uncertainties